src/components/TopicBarChart.js

```diff
@@ -1,139 +1,56 @@
-    const labelWidth = 150;
-    const innerPadding = 10;
-    const padding = 20;
-    const remainingWidth = w - (padding * 2) - (innerPadding * 2);
-    const countScale = d3.scaleLinear()
-      .domain(d3.extent(topics, d => d.count))
-      .range([5, w - labelWidth]);
-
-    const sumCounts = d3.sum(topics, d => d.count);
-
-    const binding = chart.selectAll('div.bar-group')
-      .data(topics, d => d.id);
-
-    const entering = binding.enter()
-      .append('div')
-      .classed('bar-group', true)
-      .style('width', `${remainingWidth}px`);
-
-    const labelContainer = entering.append('div')
-      .classed('topic-detail-container', true);
-
-    // name of topic
-    labelContainer.append('div')
-      .classed('topic-name', true)
-      .text(d => d.answer);
-
-    // % label
-    labelContainer.append('div')
-      .classed('percentage', true);
-
-    const barContainer = entering.append('div')
-      .classed('topic-bar-container', true);
-
-    // bar - background
-    barContainer.append('div')
-        .classed('topic-bar-background', true)
-        .style('width', `${remainingWidth}px`);
-
-    // bar - value
-    barContainer.append('div')
-      .classed('topic-bar', true);
-
-    binding.merge(entering).each(function(d) {
-      const node = d3.select(this);
-      node.selectAll('div.topic-bar')
-        .style('width', d => `${countScale(d.count)}px`);
-      node.selectAll('div.percentage', true)
-        .text(d => d3.format('.0%')(d.count / sumCounts));
-    });
-  }
-
-  render() {
-    const { width } = this.props;
+    const responseCount = d3.sum(topics, d => d.count);
 
     return (
       <div className={'topic-bar-chart'}>
         <h3>Topics</h3>
-        <p>
+        {<p>
           Lorem ipsum dolor sit amet, consectetur adipiscing elit.
           Quisque eu velit orci. Donec in aliquet nunc. Duis pretium
           vulputate nunc id elementum. Vivamus commodo eros nisl,
           vel pretium lectus ultricies id. Curabitur eu dui malesuada,
-          rutrum libero at, lacinia mauris. Quisque a fermentum nisi.
-          Praesent porta ante eu congue placerat. Aliquam semper ligula
-          x, id cursus lectus lobortis in. In hac habitasse platea dictumst.
-          Sed a ipsum ac magna volutpat laoreet. Aenean ultricies venenatis
-          velit, facilisis sollicitudin eros vehicula sed.
-        </p>
-        <div
-          className="bar-group-container"
-          style={{
-            position: 'relative',
-            width: `${width}px`
-          }}
-          ref={(node) => { this.chart = node; }}
-        />
+          rutrum libero at, lacinia mauris.
+        </p>}
+        <div className="bar-group-container" style={{ maxWidth: `${width}px` }}>
+          <small className="note">{responseCount} total responses</small>
+          {topics.map((topic) => {
+            const percentage = topic.count ?
+              d3.format('.0%')(topic.count / responseCount) :
```

Comment:
if responseCount is 0 topic count will by nature also be 0, so we'll hit the ternary